frontend/packages/core/src/dialog.tsx

```diff
@@ -1,27 +1,61 @@
-import React from "react";
-import MuiDialog from "@material-ui/core/Dialog";
-import DialogActions from "@material-ui/core/DialogActions";
-import DialogContent from "@material-ui/core/DialogContent";
-import DialogTitle from "@material-ui/core/DialogTitle";
-import Typography from "@material-ui/core/Typography";
+import * as React from "react";
+import styled from "@emotion/styled";
+import type { DialogProps as MuiDialogProps } from "@material-ui/core";
+import {
+  Dialog as MuiDialog,
+  DialogActions as MuiDialogActions,
+  DialogContent as MuiDialogContent,
+  DialogTitle as MuiDialogTitle,
+  IconButton as MuiIconButton,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
 
-export interface DialogProps {
+const DialogTitle = styled(MuiDialogTitle)({
+  display: "flex",
+  justifyContent: "space-between",
+  fontSize: "16px",
+  padding: "16px 16px 0 16px",
+  fontWeight: 400,
+  color: "#000000",
+});
+
+const IconButton = styled(MuiIconButton)({
+  height: "16px",
+  width: "16px",
+  padding: "0",
+  color: "#000000",
+});
+
+const DialogContent = styled(MuiDialogContent)({
+  padding: "8px 16px",
+  fontSize: "12px",
+  fontWeight: 400,
+  color: "rgba(13, 16, 48, 0.6)",
+});
+
+const DialogActions = styled(MuiDialogActions)({
+  borderTop: "1px solid rgba(13, 16, 48, 0.12)",
+  padding: "0 8px",
+});
+
+export interface DialogProps extends Pick<MuiDialogProps, "open"> {
   title: string;
-  content: string;
-  open: boolean;
+  children: React.ReactNode;
+  actions: React.ReactNode;
```

Comment:
we should standardize, either use these richer API where you pass components to be rendered nested vs just using children for everything and passing in Actions as part of the children (this same thing applies for accordion). i think we just need to some quick pro/con on it.

my main thought in potentially choosing the children-only API is that it reads more literally despite being a little more complicated for the user (but that's what stories are for, or maybe an invariant check on top of it), i.e. title > content > buttons, whereas this API reads title > buttons > content but renders as title > content > buttons